src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2809,6 +2809,8 @@ private UnboundLambda GetUnboundLambda(BoundLambda expr, VariableState variableS
         private MethodSymbol InferMethodTypeArguments(BoundCall node, MethodSymbol method, ImmutableArray<BoundExpression> arguments)
         {
             Debug.Assert(method.IsGenericMethod);
+            Debug.Assert(arguments.All(a => a.GetTypeAndNullability().NullableAnnotation.IsSpeakable()));
```

Comment:
If all arguments are expressions with speakable annotations, that suggests that only speakable annotations are used at all (declared annotations are always speakable, and these are the flow states. Those are the only things annotations are used for).  So why do we have the other states?